### PR TITLE
[plugins/aws][feat] Add edge between EC2 Snapshots, Volumes and KMS Keys

### PR DESCRIPTION
**Pull request by anjafr**
# Description

Add edge between EC2 Snapshots, Volumes and KMS Keys

# To-Dos

- [x] Lint and test with `tox`

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

File: plugins/aws/resoto_plugin_aws/resource/ec2.py [hunk 1]
```diff
@@ -525,6 +534,12 @@ def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
         super().connect_in_graph(builder, source)
         if volume_id := source.get("VolumeId"):
             builder.add_edge(self, EdgeType.default, reverse=True, clazz=AwsEc2Volume, id=volume_id)
+        if self.snapshot_kms_key_id:
+            builder.dependant_node(
```
Comment (meln1k) [inline plugins/aws/resoto_plugin_aws/resource/ec2.py:538]: the same as above

File: plugins/aws/resoto_plugin_aws/resource/ec2.py [hunk 2]
```diff
@@ -471,6 +472,12 @@ def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
         builder.add_edge(self, EdgeType.default, reverse=True, name=self.volume_type)
         for attachment in self.volume_attachments:
             builder.dependant_node(self, reverse=True, clazz=AwsEc2Instance, id=attachment.instance_id)
+        if self.volume_kms_key_id:
+            builder.dependant_node(
```
Comment (meln1k) [inline plugins/aws/resoto_plugin_aws/resource/ec2.py:476]: should it be a reverse node?

Comment (anjafr) [reply plugins/aws/resoto_plugin_aws/resource/ec2.py:476]: I don't see how the key could be the leading entity here 🤔 Any reason in particular?

**Review by aquamatthias — APPROVED** [review]
thanks!

**Pull request merged**